In [94]:
import pandas as pd
raw_data=pd.read_csv("data/data.csv")
predict_data = data[pd.isna(data["x_e_out [-]"])]
predict_data=predict_data.drop(columns=["x_e_out [-]"])
data=raw_data.copy()

In [95]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 31644 entries, 0 to 31643
Data columns (total 10 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   id                   31644 non-null  int64  
 1   author               26620 non-null  object 
 2   geometry             26144 non-null  object 
 3   pressure [MPa]       27192 non-null  float64
 4   mass_flux [kg/m2-s]  26853 non-null  float64
 5   x_e_out [-]          21229 non-null  float64
 6   D_e [mm]             26156 non-null  float64
 7   D_h [mm]             27055 non-null  float64
 8   length [mm]          26885 non-null  float64
 9   chf_exp [MW/m2]      31644 non-null  float64
dtypes: float64(7), int64(1), object(2)
memory usage: 2.4+ MB


In [96]:
data.head()

id    author geometry  pressure [MPa]  mass_flux [kg/m2-s]  x_e_out [-]  \
0   0  Thompson     tube            7.00               3770.0       0.1754   
1   1  Thompson     tube             NaN               6049.0      -0.0416   
2   2  Thompson      NaN           13.79               2034.0       0.0335   
3   3      Beus  annulus           13.79               3679.0      -0.0279   
4   4       NaN     tube           13.79                686.0          NaN   

   D_e [mm]  D_h [mm]  length [mm]  chf_exp [MW/m2]  
0       NaN      10.8        432.0              3.6  
1      10.3      10.3        762.0              6.2  
2       7.7       7.7        457.0              2.5  
3       5.6      15.2       2134.0              3.0  
4      11.1      11.1        457.0              2.8

In [103]:
print(data["author"].unique())
print(data["geometry"].unique())
author_geometry=data[["author","geometry"]].drop_duplicates().dropna()
author_geometry

['Thompson' 'Beus' nan 'Peskov' 'Janssen' 'Weatherhead' 'Inasaka'
 'Williams' 'Mortimore' 'Richenderfer' 'Kossolapov']
['tube' nan 'annulus' 'plate']


author geometry
0          Thompson     tube
3              Beus  annulus
7            Peskov     tube
11          Janssen  annulus
22      Weatherhead     tube
40          Inasaka     tube
57         Williams     tube
77        Mortimore  annulus
108    Richenderfer    plate
235      Kossolapov    plate
693    Richenderfer     tube
809            Beus     tube
934       Mortimore     tube
2664   Richenderfer  annulus
3180         Peskov    plate
4476       Thompson    plate
5106         Peskov  annulus
5399        Janssen     tube
6412       Williams    plate
7137       Thompson  annulus
12514    Kossolapov     tube
15910     Mortimore    plate
16102       Janssen    plate
18026      Williams  annulus
23122    Kossolapov  annulus
23470       Inasaka    plate
26769   Weatherhead  annulus

In [114]:
dic_autor_geometry = {}
for _, row in author_geometry.iterrows():
    autor = row['author']
    geometry = row['geometry']
    if autor in dic_autor_geometry:
        dic_autor_geometry[autor].append(geometry)
    else:
        dic_autor_geometry[autor] = [geometry]

# Imprimir el diccionario
print(dic_autor_geometry)

{'Thompson': ['tube', 'plate', 'annulus'], 'Beus': ['annulus', 'tube'], 'Peskov': ['tube', 'plate', 'annulus'], 'Janssen': ['annulus', 'tube', 'plate'], 'Weatherhead': ['tube', 'annulus'], 'Inasaka': ['tube', 'plate'], 'Williams': ['tube', 'plate', 'annulus'], 'Mortimore': ['annulus', 'tube', 'plate'], 'Richenderfer': ['plate', 'tube', 'annulus'], 'Kossolapov': ['plate', 'tube', 'annulus']}


In [115]:
for autor in dic_autor_geometry:
    for geometria in dic_autor_geometry[autor]:
        print(autor, geometria)
        print(data[data["author"]==autor][data["geometry"]==geometria].describe())

Thompson tube
                 id  pressure [MPa]  mass_flux [kg/m2-s]  x_e_out [-]  \
count  14324.000000    12393.000000         12180.000000   9555.00000   
mean   15877.367565       10.660803          3496.055583     -0.01384   
std     9158.588745        4.276420          1838.097515      0.10361   
min        0.000000        0.100000             0.000000     -0.86670   
25%     7915.000000        6.890000          1980.000000     -0.05940   
50%    15914.500000       11.030000          3662.000000     -0.00780   
75%    23791.250000       13.790000          4136.000000      0.05330   
max    31642.000000       20.680000          7975.000000      0.21200   

           D_e [mm]      D_h [mm]   length [mm]  chf_exp [MW/m2]  
count  11831.000000  12270.000000  12229.000000     14324.000000  
mean       8.210692      8.270326    609.942595         4.138362  
std        5.883229      6.116300    464.029507         2.000802  
min        1.000000      1.000000     10.000000         0.80

c:\Users\mgilj\AppData\Local\Programs\Python\Python37\lib\site-packages\ipykernel_launcher.py:4: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  after removing the cwd from sys.path.
c:\Users\mgilj\AppData\Local\Programs\Python\Python37\lib\site-packages\ipykernel_launcher.py:4: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  after removing the cwd from sys.path.
c:\Users\mgilj\AppData\Local\Programs\Python\Python37\lib\site-packages\ipykernel_launcher.py:4: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  after removing the cwd from sys.path.
c:\Users\mgilj\AppData\Local\Programs\Python\Python37\lib\site-packages\ipykernel_launcher.py:4: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  after removing the cwd from sys.path.


                 id  pressure [MPa]  mass_flux [kg/m2-s]  x_e_out [-]  \
count   1308.000000     1110.000000          1119.000000   881.000000   
mean   15949.391437       13.065667          2350.529044     0.060299   
std     9106.195292        2.472537          1055.863033     0.082026   
min        3.000000        2.140000           671.000000    -0.313700   
25%     8087.500000       11.030000          1355.000000     0.010100   
50%    15878.500000       13.790000          2710.000000     0.056800   
75%    23993.500000       15.510000          3665.000000     0.123300   
max    31608.000000       18.270000          6200.000000     0.232000   

          D_e [mm]     D_h [mm]  length [mm]  chf_exp [MW/m2]  
count  1065.000000  1097.000000  1115.000000      1308.000000  
mean      5.611549    15.102097  2132.222422         2.230734  
std       0.301356     1.022692    51.037977         1.101222  
min       5.600000     5.600000   457.000000         0.800000  
25%       5.600000    

c:\Users\mgilj\AppData\Local\Programs\Python\Python37\lib\site-packages\ipykernel_launcher.py:4: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  after removing the cwd from sys.path.
c:\Users\mgilj\AppData\Local\Programs\Python\Python37\lib\site-packages\ipykernel_launcher.py:4: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  after removing the cwd from sys.path.
c:\Users\mgilj\AppData\Local\Programs\Python\Python37\lib\site-packages\ipykernel_launcher.py:4: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  after removing the cwd from sys.path.
c:\Users\mgilj\AppData\Local\Programs\Python\Python37\lib\site-packages\ipykernel_launcher.py:4: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  after removing the cwd from sys.path.


           id  pressure [MPa]  mass_flux [kg/m2-s]  x_e_out [-]  D_e [mm]  \
count     1.0             1.0                  1.0       1.0000       1.0   
mean   5106.0            10.0               2200.0       0.1344      10.0   
std       NaN             NaN                  NaN          NaN       NaN   
min    5106.0            10.0               2200.0       0.1344      10.0   
25%    5106.0            10.0               2200.0       0.1344      10.0   
50%    5106.0            10.0               2200.0       0.1344      10.0   
75%    5106.0            10.0               2200.0       0.1344      10.0   
max    5106.0            10.0               2200.0       0.1344      10.0   

       D_h [mm]  length [mm]  chf_exp [MW/m2]  
count       1.0          1.0              1.0  
mean       10.0        893.0              1.1  
std         NaN          NaN              NaN  
min        10.0        893.0              1.1  
25%        10.0        893.0              1.1  
50%        10.0   

c:\Users\mgilj\AppData\Local\Programs\Python\Python37\lib\site-packages\ipykernel_launcher.py:4: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  after removing the cwd from sys.path.
c:\Users\mgilj\AppData\Local\Programs\Python\Python37\lib\site-packages\ipykernel_launcher.py:4: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  after removing the cwd from sys.path.
c:\Users\mgilj\AppData\Local\Programs\Python\Python37\lib\site-packages\ipykernel_launcher.py:4: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  after removing the cwd from sys.path.
c:\Users\mgilj\AppData\Local\Programs\Python\Python37\lib\site-packages\ipykernel_launcher.py:4: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  after removing the cwd from sys.path.


                 id  pressure [MPa]  mass_flux [kg/m2-s]  x_e_out [-]  \
count   1698.000000     1448.000000           1440.00000  1146.000000   
mean   15867.604829       13.773322           1549.51250    -0.014685   
std     9245.660125        0.433456            686.75167     0.112522   
min       22.000000        4.140000            332.00000    -0.485400   
25%     7662.250000       13.790000           1301.75000    -0.073675   
50%    15948.000000       13.790000           1370.00000     0.004450   
75%    23945.750000       13.790000           1994.00000     0.060050   
max    31637.000000       18.960000           7500.00000     0.194600   

          D_e [mm]     D_h [mm]  length [mm]  chf_exp [MW/m2]  
count  1396.000000  1448.000000  1456.000000      1698.000000  
mean      8.983811     8.994751   458.504808         2.964252  
std       1.912594     1.895304    77.077896         1.674869  
min       1.900000     1.900000    10.000000         0.800000  
25%       7.700000    

c:\Users\mgilj\AppData\Local\Programs\Python\Python37\lib\site-packages\ipykernel_launcher.py:4: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  after removing the cwd from sys.path.
c:\Users\mgilj\AppData\Local\Programs\Python\Python37\lib\site-packages\ipykernel_launcher.py:4: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  after removing the cwd from sys.path.
c:\Users\mgilj\AppData\Local\Programs\Python\Python37\lib\site-packages\ipykernel_launcher.py:4: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  after removing the cwd from sys.path.


            id  pressure [MPa]  mass_flux [kg/m2-s]  x_e_out [-]  D_e [mm]  \
count      1.0            1.00                  1.0          0.0       1.0   
mean   23470.0            0.39                706.0          NaN      15.0   
std        NaN             NaN                  NaN          NaN       NaN   
min    23470.0            0.39                706.0          NaN      15.0   
25%    23470.0            0.39                706.0          NaN      15.0   
50%    23470.0            0.39                706.0          NaN      15.0   
75%    23470.0            0.39                706.0          NaN      15.0   
max    23470.0            0.39                706.0          NaN      15.0   

       D_h [mm]  length [mm]  chf_exp [MW/m2]  
count       1.0          1.0              1.0  
mean      120.0         10.0              6.5  
std         NaN          NaN              NaN  
min       120.0         10.0              6.5  
25%       120.0         10.0              6.5  
50%      

c:\Users\mgilj\AppData\Local\Programs\Python\Python37\lib\site-packages\ipykernel_launcher.py:4: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  after removing the cwd from sys.path.
c:\Users\mgilj\AppData\Local\Programs\Python\Python37\lib\site-packages\ipykernel_launcher.py:4: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  after removing the cwd from sys.path.
c:\Users\mgilj\AppData\Local\Programs\Python\Python37\lib\site-packages\ipykernel_launcher.py:4: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  after removing the cwd from sys.path.
c:\Users\mgilj\AppData\Local\Programs\Python\Python37\lib\site-packages\ipykernel_launcher.py:4: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  after removing the cwd from sys.path.
c:\Users\mgilj\AppData\Local\Programs\Python\Python37\lib\site-packages\ipykernel_launcher.py:4: UserWarning: Boolean Series key will be reindexed t

            id  pressure [MPa]  mass_flux [kg/m2-s]  x_e_out [-]  D_e [mm]  \
count      1.0            1.00                  1.0       1.0000       0.0   
mean   18026.0           15.17               2699.0       0.0486       NaN   
std        NaN             NaN                  NaN          NaN       NaN   
min    18026.0           15.17               2699.0       0.0486       NaN   
25%    18026.0           15.17               2699.0       0.0486       NaN   
50%    18026.0           15.17               2699.0       0.0486       NaN   
75%    18026.0           15.17               2699.0       0.0486       NaN   
max    18026.0           15.17               2699.0       0.0486       NaN   

       D_h [mm]  length [mm]  chf_exp [MW/m2]  
count       1.0          1.0              1.0  
mean        9.5       1836.0              1.5  
std         NaN          NaN              NaN  
min         9.5       1836.0              1.5  
25%         9.5       1836.0              1.5  
50%      

c:\Users\mgilj\AppData\Local\Programs\Python\Python37\lib\site-packages\ipykernel_launcher.py:4: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  after removing the cwd from sys.path.
c:\Users\mgilj\AppData\Local\Programs\Python\Python37\lib\site-packages\ipykernel_launcher.py:4: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  after removing the cwd from sys.path.
c:\Users\mgilj\AppData\Local\Programs\Python\Python37\lib\site-packages\ipykernel_launcher.py:4: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  after removing the cwd from sys.path.
c:\Users\mgilj\AppData\Local\Programs\Python\Python37\lib\site-packages\ipykernel_launcher.py:4: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  after removing the cwd from sys.path.
c:\Users\mgilj\AppData\Local\Programs\Python\Python37\lib\site-packages\ipykernel_launcher.py:4: UserWarning: Boolean Series key will be reindexed t

                 id  pressure [MPa]  mass_flux [kg/m2-s]  x_e_out [-]  \
count    408.000000      353.000000           338.000000   283.000000   
mean   16264.495098        0.850057          1622.511834    -0.029778   
std     8955.568201        1.987868          1097.595987     0.038195   
min      108.000000        0.100000            10.000000    -0.148600   
25%     8640.500000        0.100000          1000.000000    -0.043300   
50%    16549.500000        0.200000          1500.000000    -0.020400   
75%    23853.000000        0.510000          2000.000000    -0.010200   
max    31634.000000       18.000000          6400.000000     0.138300   

         D_e [mm]    D_h [mm]  length [mm]  chf_exp [MW/m2]  
count  336.000000  350.000000   344.000000       408.000000  
mean    14.865476  117.827143    21.610465         5.363480  
std      1.107235   15.245687   105.651937         1.845937  
min      1.900000    1.900000    10.000000         1.800000  
25%     15.000000  120.000000   

c:\Users\mgilj\AppData\Local\Programs\Python\Python37\lib\site-packages\ipykernel_launcher.py:4: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  after removing the cwd from sys.path.


Relaciones encontradas

Mortimore annulus
    length [mm]=2134


Mortimore tube
    length [mm]=2134.0


Richenderfer tube
    D_e [mm]=15
    D_h [mm]=120



Richenderfer annulus
    D_e [mm] = 15
    D_h [mm]=120
    length [mm]=10


Kossolapov tube
    D_e [mm]=15
    D_h [mm]=120
    length [mm]=10



In [117]:
data["author"].unique()

array(['Thompson', 'Beus', nan, 'Peskov', 'Janssen', 'Weatherhead',
       'Inasaka', 'Williams', 'Mortimore', 'Richenderfer', 'Kossolapov'],
      dtype=object)

In [119]:
original_data=pd.read_csv("data/original_dataset.csv")

In [121]:
original_data.head()

id   author geometry  pressure [MPa]  mass_flux [kg/m2-s]  x_e_out [-]  \
0   1  Inasaka     tube            0.39                 5600      -0.1041   
1   2  Inasaka     tube            0.31                 6700      -0.0596   
2   3  Inasaka     tube            0.33                 4300      -0.0395   
3   4  Inasaka     tube            0.62                 6400      -0.1460   
4   5  Inasaka     tube            0.64                 4700      -0.0849   

   D_e [mm]  D_h [mm]  length [mm]  chf_exp [MW/m2]  
0       3.0       3.0          100             11.3  
1       3.0       3.0          100             10.6  
2       3.0       3.0          100              7.3  
3       3.0       3.0          100             12.8  
4       3.0       3.0          100             11.0

In [122]:
original_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1865 entries, 0 to 1864
Data columns (total 10 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   id                   1865 non-null   int64  
 1   author               1865 non-null   object 
 2   geometry             1865 non-null   object 
 3   pressure [MPa]       1865 non-null   float64
 4   mass_flux [kg/m2-s]  1865 non-null   int64  
 5   x_e_out [-]          1865 non-null   float64
 6   D_e [mm]             1865 non-null   float64
 7   D_h [mm]             1865 non-null   float64
 8   length [mm]          1865 non-null   int64  
 9   chf_exp [MW/m2]      1865 non-null   float64
dtypes: float64(5), int64(3), object(2)
memory usage: 145.8+ KB


In [126]:
original_data[["author","geometry"]].drop_duplicates()

author geometry
0          Inasaka     tube
7           Peskov     tube
24        Thompson     tube
1226   Weatherhead     tube
1388      Williams     tube
1439          Beus  annulus
1516       Janssen  annulus
1798     Mortimore  annulus
1817    Kossolapov    plate
1829  Richenderfer    plate

In [131]:
original_data[original_data["author"]=="Inasaka"]

id   author geometry  pressure [MPa]  mass_flux [kg/m2-s]  x_e_out [-]  \
0   1  Inasaka     tube            0.39                 5600      -0.1041   
1   2  Inasaka     tube            0.31                 6700      -0.0596   
2   3  Inasaka     tube            0.33                 4300      -0.0395   
3   4  Inasaka     tube            0.62                 6400      -0.1460   
4   5  Inasaka     tube            0.64                 4700      -0.0849   
5   6  Inasaka     tube            0.64                 6200      -0.1221   
6   7  Inasaka     tube            0.91                 5500      -0.1224   

   D_e [mm]  D_h [mm]  length [mm]  chf_exp [MW/m2]  
0       3.0       3.0          100             11.3  
1       3.0       3.0          100             10.6  
2       3.0       3.0          100              7.3  
3       3.0       3.0          100             12.8  
4       3.0       3.0          100             11.0  
5       3.0       3.0          100              8.2  
6       3.0       3.0          100             15.6

In [132]:
data[data["author"]=="Inasaka"]

id   author geometry  pressure [MPa]  mass_flux [kg/m2-s]  \
40        40  Inasaka     tube           12.07               6200.0   
530      530  Inasaka     tube            0.64               6700.0   
1100    1100  Inasaka     tube            0.39                  NaN   
1610    1610  Inasaka      NaN            0.64               6700.0   
1731    1731  Inasaka      NaN            0.33                  NaN   
2644    2644  Inasaka      NaN            0.91               6700.0   
4293    4293  Inasaka     tube            0.33               4300.0   
6391    6391  Inasaka     tube            0.62               6200.0   
6550    6550  Inasaka     tube            0.91               4700.0   
6729    6729  Inasaka     tube            0.91                  NaN   
6888    6888  Inasaka     tube             NaN               5500.0   
8093    8093  Inasaka     tube           18.27                842.0   
8513    8513  Inasaka     tube             NaN                  NaN   
8598    8598  Inasaka     tube             NaN               6400.0   
10384  10384  Inasaka     tube            0.31               6700.0   
10697  10697  Inasaka     tube            0.39               4300.0   
11688  11688  Inasaka     tube            0.51               5500.0   
12336  12336  Inasaka      NaN            0.39               6200.0   
13349  13349  Inasaka     tube            0.64               4700.0   
13371  13371  Inasaka      NaN            0.64               6200.0   
13709  13709  Inasaka     tube           10.51               5600.0   
14082  14082  Inasaka     tube            0.10               4300.0   
14522  14522  Inasaka     tube            3.45               4700.0   
15667  15667  Inasaka     tube            0.64               4300.0   
16329  16329  Inasaka     tube            0.33               4300.0   
16634  16634  Inasaka     tube            0.33                  NaN   
18367  18367  Inasaka     tube            0.39               5600.0   
18687  18687  Inasaka     tube             NaN               5500.0   
20512  20512  Inasaka     tube            0.51               4300.0   
21648  21648  Inasaka     tube            0.91               6400.0   
22147  22147  Inasaka      NaN            0.62               6400.0   
22447  22447  Inasaka     tube            0.62               5361.0   
23031  23031  Inasaka     tube            0.33               4700.0   
23470  23470  Inasaka    plate            0.39                706.0   
23476  23476  Inasaka     tube           12.41               4986.0   
24166  24166  Inasaka      NaN            0.51               4700.0   
24501  24501  Inasaka      NaN           10.43               3065.0   
24620  24620  Inasaka     tube            0.31               6700.0   
24705  24705  Inasaka      NaN            0.51               5600.0   
27520  27520  Inasaka      NaN            0.10               6700.0   
27760  27760  Inasaka     tube            0.31               6200.0   
28962  28962  Inasaka     tube            0.64                  NaN   
29640  29640  Inasaka      NaN            0.64               6200.0   
29674  29674  Inasaka     tube           13.79               4700.0   
29990  29990  Inasaka     tube            0.64                  NaN   
30022  30022  Inasaka     tube            0.33               6700.0   

       x_e_out [-]  D_e [mm]  D_h [mm]  length [mm]  chf_exp [MW/m2]  
40         -0.0692       3.0       NaN        100.0              8.0  
530            NaN       3.0       3.0        100.0              7.4  
1100           NaN       3.0       3.0         10.0              8.2  
1610       -0.1041       3.0       NaN        100.0              8.1  
1731       -0.1221       3.0       3.0        100.0              8.2  
2644           NaN       3.0       3.0        100.0              8.2  
4293           NaN       NaN       3.0        100.0              8.4  
6391       -0.1460       3.0       3.0        100.0              7.9  
6550           NaN       3.0     

In [129]:
author_geometry.drop_duplicates()

author geometry
0          Thompson     tube
3              Beus  annulus
7            Peskov     tube
11          Janssen  annulus
22      Weatherhead     tube
40          Inasaka     tube
57         Williams     tube
77        Mortimore  annulus
108    Richenderfer    plate
235      Kossolapov    plate
693    Richenderfer     tube
809            Beus     tube
934       Mortimore     tube
2664   Richenderfer  annulus
3180         Peskov    plate
4476       Thompson    plate
5106         Peskov  annulus
5399        Janssen     tube
6412       Williams    plate
7137       Thompson  annulus
12514    Kossolapov     tube
15910     Mortimore    plate
16102       Janssen    plate
18026      Williams  annulus
23122    Kossolapov  annulus
23470       Inasaka    plate
26769   Weatherhead  annulus

In [127]:
original_data["geometry"].unique()

array(['tube', 'annulus', 'plate'], dtype=object)